inc/classes/subscriber/Tools/class-cloudflare-subscriber.php

```diff
@@ -0,0 +1,888 @@
+
+		if ( is_wp_error( $cf_cache_everything ) || ! $cf_cache_everything ) {
+			return;
+		}
+
+		// Purge CloudFlare.
+		$cf_purge = $this->purge_cloudflare();
+
+		if ( is_wp_error( $cf_purge ) ) {
+			$cf_purge_result = [
+				'result'  => 'error',
+				// translators: %s = CloudFare API return message.
+				'message' => sprintf( __( 'Cloudflare cache purge error: %s', 'rocket' ), $cf_purge->get_error_message() ),
+			];
+		} else {
+			$cf_purge_result = [
+				'result'  => 'success',
+				'message' => __( 'Cloudflare cache successfully purged', 'rocket' ),
+			];
+		}
+
+		set_transient( get_current_user_id() . '_cloudflare_purge_result', $cf_purge_result );
+	}
+
+	/**
+	 * Purge Cloudflare cache URLs automatically if Cache Everything is set as a Page Rule
+	 *
+	 * @since 3.4.2
+	 * @author Soponar Cristina
+	 *
+	 * @param WP_Post $post       The post object.
+	 * @param array   $purge_urls URLs cache files to remove.
+	 * @param string  $lang       The post language.
+	 */
+	public function auto_purge_by_url( $post, $purge_urls, $lang ) {
+		if ( ! get_rocket_option( 'do_cloudflare' ) || ! current_user_can( 'rocket_purge_cloudflare_cache' ) ) {
+			return;
+		}
+
+		$cf_cache_everything = $this->has_page_rule( 'cache_everything' );
+
+		if ( is_wp_error( $cf_cache_everything ) || ! $cf_cache_everything ) {
+			return;
+		}
+
+		// Purge CloudFlare.
+		$cf_purge = $this->purge_by_url( $post, $purge_urls, $lang );
+
+		if ( is_wp_error( $cf_purge ) ) {
+			$cf_purge_result = [
+				'result'  => 'error',
+				// translators: %s = CloudFare API return message.
+				'message' => sprintf( __( 'Cloudflare cache purge error: %s', 'rocket' ), $cf_purge->get_error_message() ),
+			];
+		} else {
+			$cf_purge_result = [
+				'result'  => 'success',
+				'message' => __( 'Cloudflare cache successfully purged', 'rocket' ),
+			];
+		}
+
+		set_transient( get_current_user_id() . '_cloudflare_purge_result', $cf_purge_result );
+	}
+
+	/**
+	 * Purge CloudFlare cache
+	 *
+	 * @since 2.5
+	 */
+	public function do_purge_cloudflare() {
+		if ( ! isset( $_GET['_wpnonce'] ) || ! wp_verify_nonce( $_GET['_wpnonce'], 'rocket_purge_cloudflare' ) ) {
+			wp_nonce_ays( '' );
+		}
+
+		if ( ! current_user_can( 'rocket_purge_cloudflare_cache' ) ) {
+			return;
+		}
+
+		// Purge CloudFlare.
+		$cf_purge = $this->purge_cloudflare();
+
+		if ( is_wp_error( $cf_purge ) ) {
+			$cf_purge_result = [
+				'result'  => 'error',
+				// translators: %s = CloudFare API return message.
+				'message' => sprintf( __( '<strong>WP Rocket:</strong> %s', 'rocket' ), $cf_purge->get_error_message() ),
+			];
+		} else {
+			$cf_purge_result = [
+				'result'  => 'success',
+				'message' => __( '<strong>WP Rocket:</strong> Cloudflare cache successfully purged.', 'rocket' ),
+			];
+		}
+
+		set_transient( get_current_user_id() . '_cloudflare_purge_result', $cf_purge_result );
+
+		wp_safe_redirect( esc_url_raw( wp_get_referer() ) );
+		die();
```

Comment:
```suggestion
		wp_die();
```

That makes the method testable, as we can mock `wp_die()`, but we can't do it for `die()`